pkg/kv/kvserver/replica_evaluate.go

```diff
@@ -333,7 +334,14 @@ func evaluateBatch(
 		if limit := baHeader.MaxSpanRequestKeys; limit > 0 {
 			retResults := reply.Header().NumKeys
 			if retResults > limit {
-				log.Fatalf(ctx, "received %d results, limit was %d", retResults, limit)
+				index, retResults, limit := index, retResults, limit // don't alloc unless branch taken
```

Comment:
@nvanbenschoten did I do this right? I checked via `go build -gcflags '-m' ./pkg/kv/kvserver/`  that the online line `index escapes to heap` refers to the locally defined index here.